galaxy/models.py

```diff
@@ -0,0 +1,379 @@
+        return 0
+
+    @classmethod
+    def compute_users_pictures_score(cls, user1, user2):
+        score = (
+            Picture.objects.filter(people__user__in=(user1,))
+            .filter(people__user__in=(user2,))
+            .count()
+        )
+        for _ in range(score):
+            print("    - '%s' was pictured with '%s'" % (user1, user2))
+        return score * cls.PICTURE_POINTS
+
+    @classmethod
+    def compute_users_clubs_score(cls, user1, user2):
+        common_clubs = Club.objects.filter(members__in=user1.memberships.all()).filter(
+            members__in=user2.memberships.all()
+        )
+        user1_memberships = user1.memberships.filter(club__in=common_clubs)
+        user2_memberships = user2.memberships.filter(club__in=common_clubs)
+
+        score = 0
+        for user1_membership in user1_memberships:
+            if user1_membership.end_date is None:
+                user1_membership.end_date = timezone.now().date()
+            print(
+                "\t- Checking range [%s, %s] for club %s"
+                % (
+                    user1_membership.start_date,
+                    user1_membership.end_date,
+                    user1_membership.club,
+                )
+            )
+            query = Q(  # start2 <= start1 <= end2
+                start_date__lte=user1_membership.start_date,
+                end_date__gte=user1_membership.start_date,
+            )
+            query |= Q(  # start2 <= start1 <= now
+                start_date__lte=user1_membership.start_date, end_date=None
+            )
+            query |= Q(  # start1 <= start2 <= end2
+                start_date__gte=user1_membership.start_date,
+                start_date__lte=user1_membership.end_date,
+            )
+            for user2_membership in user2_memberships.filter(
+                query, club=user1_membership.club
+            ):
+                if user2_membership.end_date is None:
+                    user2_membership.end_date = timezone.now().date()
+                latest_start = max(
+                    user1_membership.start_date, user2_membership.start_date
+                )
+                earliest_end = min(user1_membership.end_date, user2_membership.end_date)
+                print(
+                    "      - '%s' was with '%s' in %s starting on %s until %s (%s days)"
+                    % (
+                        user1,
+                        user2,
+                        user2_membership.club,
+                        latest_start,
+                        earliest_end,
+                        (earliest_end - latest_start).days,
+                    )
+                )
+                score += cls.CLUBS_POINTS * (earliest_end - latest_start).days
+        return score
+
+    ###################
+    # Rule the galaxy #
+    ###################
+
+    @classmethod
+    def rule(cls):
+        GalaxyStar.objects.all().delete()
+        GalaxyLane.objects.all().delete()
```

Comment:
Le modèle GalaxyLane est lié par une clef étrangère à GalaxyStar, avec on_delete=CASCADE, donc, au moment où cette requête est faite, la table GalaxyLane est déjà vide.

Reply:
Comment added to precise that it's currently a no-op, but that it's still better to do it anyway to prevent future mess.